dio/lib/src/dio_mixin.dart

```diff
@@ -337,7 +337,12 @@ abstract class DioMixin implements Dio {
     Options? options,
     ProgressCallback? onSendProgress,
     ProgressCallback? onReceiveProgress,
-  }) {
+  }) async {
+    if (cancelToken != null && cancelToken.isCancelled) {
+      final exception = await cancelToken.whenCancel;
+      throw exception;
```

Comment:
```suggestion
      throw cancelToken.cancelError!;
```

You could force cast it here since `isCancelled` checks for the existence of the `cancelError`.